### PR TITLE
fix : expanded firebase tests

### LINKED ISSUE #236: Expand Firestore security rules tests
## Problem

Rules tests do not cover all participant and role edge cases.

## What to do

- Tests: student cannot write others participating docs.
- Club cannot self-assign admin.
- Check-in write rules.

## Files

- `tests/firestore.rules.test.ts`

## Done when

- [ ] npm run test:rules passes with new cases

### PR DESCRIPTION
**Pull request by AarishMansur**
# Description
added new test cases to expand the firestore security rules coverage as mentioned in issue . the existing tests were missing several important edge cases around participant writes and role assignments which could of left security gaps unnoticed.

> **Note:**  also while setting up i found a pre-existing TypeScript error in the test file itself `claims` was typed as `object` which is not assignable to `TokenOptions` (firebase's type requires an index signature i.e `Record<string, unknown>`). this was causing the entire test suite to fail before even running. fixed that too as part of this PR.

Fixes #236

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

ran `npm run test:rules` locally with firebase emulator, all three new cases passes successfully.

- [x] student cannot write to others participant docs
- [x] club cannot self-assign admin role
- [x] check-in write rules are restricted correctly

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Summary by CodeRabbit

* **New Features**
  * Added security rules for event check-ins: authenticated users can read; create/update/delete limited to admins, club users, or the event owner. Participant rules unchanged.

* **Tests**
  * Expanded test coverage for check-in read/write cases, prevention of self-assigning elevated roles, and denial of deleting other users’ participant records.

[Review Change Stack](https://app.coderabbit.ai/change-stack/roshankumar0036singh/Uni-Event/pull/270?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -20,9 +20,18 @@ service cloud.firestore {
...
+        allow create, delete, update: if request.auth != null && request.auth.uid == participantId;
+      }
+
+      match /checkIns/{checkInId} {
+        allow read: if request.auth != null;
+        allow create, update, delete: if request.auth != null && (
+          request.auth.token.admin == true ||
+          request.auth.token.club == true ||
+          request.auth.uid == get(/databases/$(database)/documents/events/$(eventId)).data.ownerId
+        );
```
[the remark this answers is not in the document]
Comment (AarishMansur) [reply firestore.rules]: done

**Comment by roshankumar0036singh** [comment]
@AarishMansur handle the coderabbit sugesstions

**Comment by AarishMansur** [comment]
@roshankumar0036singh done it in recent commit

**Review by roshankumar0036singh — APPROVED** [review]

**Comment by AarishMansur** [comment]
@roshankumar0036singh  can u please add relevant gssoc tags

**Comment by roshankumar0036singh** [comment]
@AarishMansur tags i added a are relevant ones for points
